app/jobs/create_derivatives_job_decorator.rb

```diff
@@ -35,6 +31,21 @@ def perform(file_set, file_id, filepath = nil)
       file_set.reload
       file_set.update_index
       file_set.parent.update_index if parent_needs_reindex?(file_set)
+    rescue StandardError => e
+      if time_to_live.positive?
+        # It's possible we can recover from this, so we'll give it another go.
+        logger.warning(
+          "WARNING: FileSet ID=\"#{file_set_id}\" error for #{self.class}: #{e}. " \
+          "Retries remaining #{time_to_live - 1}."
+        )
+        CreateDerivativesJob.perform_later(file_set, file_id, filepath, time_to_live - 1)
+      else
```

Comment:
I want to make sure I'm clear on the impact of this change.  We get a total of 3 tries for this job.  There will be sending these to the retry queue because of an exception.

Is that the intention?